clients/python-wrapper/lakefs/object_io.py

```diff
@@ -221,29 +223,65 @@ def _extract_etag_from_response(headers) -> str:
         etag = headers.get("ETag", "").strip(' "')
         return etag
 
-    def _upload(self, content, pre_sign, content_type: Optional[str] = None,
-                metadata: Optional[dict[str, str]] = None) -> ObjectStats:
-        if not pre_sign:
-            raise UnsupportedOperationException("Upload currently supported only in pre-sign mode")
-
-        headers = {}
-        if content_type is not None:
-            headers["Content-Type"] = content_type
+    def _upload_raw(self, content, content_type, metadata):
+        """
+        Preform upload api using raw api call to bypass validation of content parameter
+        """
+        headers = {
+            "Accept": self._client.sdk_client.objects_api.api_client.select_header_accept(['application/json']),
+            "Content-Type": content_type if content_type is not None else "application/octet-stream"
+        }
+
+        # Create user metadata headers
+        for k, v in metadata.items():
+            headers[_LAKEFS_METADATA_PREFIX + k] = v
+
+        _response_types_map = {
+            '201': "ObjectStats",
+            '400': "Error",
+            '401': "Error",
+            '403': "Error",
+            '404': "Error",
+            '412': "Error",
+            '420': None,
+        }
+        # authentication setting
+        _auth_settings = ['basic_auth', 'cookie_auth', 'oidc_auth', 'saml_auth', 'jwt_token']
+        return self._client.sdk_client.objects_api.api_client.call_api(
+            resource_path='/repositories/{repository}/branches/{branch}/objects',
+            method='POST',
+            path_params={"repository": self._repo, "branch": self._ref},
+            query_params={"path": self._path},
+            header_params=headers,
+            body=content,
+            response_types_map=_response_types_map,
+            auth_settings=_auth_settings,
+            _return_http_data_only=True
+        )
+
+    def _upload_presign(self, content, content_type: Optional[str] = None,
+                        metadata: Optional[dict[str, str]] = None) -> ObjectStats:
+        headers = {
+            "Accept": self._client.sdk_client.objects_api.api_client.select_header_accept(['application/json']),
```

Comment:
Reading the code for select_header_accept, I'm not sure why we don't just say application/json. Also this is a long way to get to the client, maybe just add a method to LakeFSClient to do this? Because we've just proven that is useful!

Reply:
1. Done
2. Adding a method to LakeFSClient good be good for the future, right now it will limit us in the lakefs_sdk version we can start using the wrapper from and slow our development process. I suggest to defer it to an improvement task

Reply:
Looking at the implementation of `select_header_accept`, this is literally:
```suggestion
            "Accept": "application/json"
```

Reply:
Sorry - it got reverted in the process